src/OptionalDeep.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Rapidez\BladeDirectives;
+
+use ArrayAccess;
+use ArrayObject;
+use Countable;
+use Illuminate\Support\Traits\Macroable;
+use IteratorAggregate;
+use JsonSerializable;
+use Traversable;
+
+class OptionalDeep implements ArrayAccess, IteratorAggregate, Countable, JsonSerializable
+{
+    use Macroable {
+        __call as macroCall;
+    }
+
+    protected $value;
+
+    public function __construct($value)
+    {
+        $this->value = $value;
+    }
+
+    public function __get($key): static
+    {
+        return new static($this->get($key));
+    }
+
+    public function get($key = null, $default = null): mixed
+    {
+        return data_get($this->value, $key, $default);
+    }
+
+    public function __set($key, $value): void
+    {
+        if (!is_object($this->value) && !is_array($this->value)) {
+            return;
+        }
+
+        $unwrappedValue = $value instanceof static ? $value->value : $value;
+
+        data_set($this->value, $key, $unwrappedValue);
+    }
+
+    public function __unset($key): void
+    {
+        if (!is_object($this->value) && !is_array($this->value)) {
+            return;
+        }
+
+        data_forget($this->value, $key);
+    }
+
+    public function __toString(): string
+    {
+        if (!isset($this->value)) {
+            return '';
+        }
+
+        if (is_string($this->value)) {
+            return $this->value;
+        }
+
+        return strval($this->value);
+    }
+
+    public function __invoke($default = null): mixed
+    {
+        return $this->get(null, $default);
+    }
+
+    public function __isset($key)
+    {
+        return isset($this->value->{$key});
+    }
+
+    public function isset(): bool
+    {
+        return isset($this->value);
+    }
+
+    public function __isNotEmpty(): bool
+    {
+        if (method_exists($this->value, 'value')) {
+            return boolval($this->value->value());
+        }
```

Comment:
Maybe it's an idea to auto-call `->value()` more often when possible? I don't know if there are many (if any) cases where this breaks functionality but it could be a nice optimization for later.